src/lib/integration_test_cloud_engine/kubernetes_network.ml

```diff
@@ -313,6 +338,103 @@ module Node = struct
     [%log info] "Sent payment"
       ~metadata:[("user_command_id", `String user_cmd_id)] ;
     ()
+
+  let dump_archive_data ~logger (t : t) ~data_file =
+    let open Malleable_error.Let_syntax in
+    let%map data =
+      Deferred.bind ~f:Malleable_error.return
+        (run_in_postgresql_container t ~n:1
+           ~cmd:
+             "pg_dump --create --no-owner \
+              postgres://postgres:foobar@localhost:5432/archive")
+    in
+    [%log info] "Dumping archive data to file %s" data_file ;
+    Out_channel.with_file data_file ~f:(fun out_ch ->
+        Out_channel.output_string out_ch data )
+
+  let dump_container_logs ~logger (t : t) ~log_file =
```

Comment:
I wrote this first, to see if I could retrieve logs. Left it in as possibly useful.